### PR TITLE
feat: default type for order channel

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: src/channeldef.ts [hunk 1]
```diff
@@ -280,7 +280,7 @@ export interface TypeMixins<T extends Type> {
    * The type of measurement (`"quantitative"`, `"temporal"`, `"ordinal"`, or `"nominal"`) for the encoded field or constant value (`datum`).
    * It can also be a `"geojson"` type for encoding ['geoshape'](https://vega.github.io/vega-lite/docs/geoshape.html).
    *
-   * Since Vega-Lite 4.14.0, Vega-Lite automatically infers data types in many cases as discussed below. However, type is required for a field if:
```
Comment (kanitw) [inline src/channeldef.ts:283]: Why removing this clause?

File: src/channeldef.ts [hunk 2]
```diff
@@ -911,6 +911,9 @@ export function defaultType<T extends TypedFieldDef<Field>>(fieldDef: T, channel
     case 'shape':
     case 'strokeDash':
       return 'nominal';
+
+    case 'order':
```
Comment (kanitw) [inline src/channeldef.ts:915]: Add comment to explain why we need to add this ordinal thing? 

(Why does it matter? Does it solve any issue?)

File: src/channeldef.ts (hunk 1 above)
Comment (domoritz) [reply src/channeldef.ts:283]: We don't mention specific versions in other places, do we?

File: src/channeldef.ts (hunk 2 above)
Comment (domoritz) [reply src/channeldef.ts:915]: Answered offline: the default was nominal but nominal causes a warning.

**Pull request merged**